src/devices/came_top432.c

```diff
@@ -0,0 +1,169 @@
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/*
+    Use this as a starting point for a new decoder.
+
+    Keep the Doxygen (slash-star-star) comment above to document the file and copyright.
+
+    Keep the Doxygen (slash-star-star) comment below to describe the decoder.
+    See http://www.doxygen.nl/manual/markdown.html for the formating options.
+
+    Remove all other multiline (slash-star) comments.
+    Use single-line (slash-slash) comments to annontate important lines if needed.
+*/
+
+/**
+(this is a markdown formatted section to describe the decoder)
+(describe the modulation, timing, and transmission, e.g.)
+The device uses PPM encoding,
+- 0 is encoded as 320 us gap and 640 us pulse,
+- 1 is encoded as 640 us gap and 320 us pulsep.
+The device sends a 4 times the packet when a button on the remote control is pressed.
+A transmission starts with a 320 us pulse. At the end of the packet, there is 36 periods of 320us between messages (11520us)
+
+Diagram :
+          |        <---> logical 0          <---> logical 1                     <-- 11520us -->
+          |       _   __    _   __   __   __    _    _    _   __   __    _    _                 _
+Pulse/gap | _____| |_|  |__| |_|  |_|  |_|  |__| |__| |__| |_|  |_|  |__| |__| |_____(..)______| |
+          +------------------------------------------------------------------------------   
+bits              ~~ : start bit (320us)
+                    |  0| 1  |  0 |  0 |  0 |  1 |  1 |  1 |  0 |  0 |  1 |  1 |
+                    |        0x4       |       0x7         |       0x3         |
+
+Data layout:
+    ccc
+- c : 4-bit data
+
+There is no CRC, no parity, no preamble (only the start pulse indicate the begining of packet)
+*/
+#include "decoder.h"
+
+#define CAMETOP432_BITLEN      12
+
+static int came_top432_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    int r; // a row index
+    uint8_t *b; // bits of a row
+    int parity;
+    uint8_t r_crc, c_crc;
+    uint16_t sensor_id;
+    uint8_t msg_type;
+    int16_t code;
+    char code_str[3];
+    
+    if (decoder->verbose > 1) {
+        bitbuffer_printf(bitbuffer, "%s: ", __func__);
+    }
+
+    for (r = 0; r < bitbuffer->num_rows; ++r) {
+        b = bitbuffer->bb[r];
+
+        /*
+         * Validate message and reject invalid messages as
+         * early as possible before attempting to parse data.
+         *
+         * Check "message envelope"
+         * - valid message length (use a minimum length to account
+         *   for stray bits appended or prepended by the demod)
+         * - valid preamble/device type/fixed bits if any
+         * - Data integrity checks (CRC/Checksum/Parity)
+         */
+
+        if (bitbuffer->bits_per_row[r] < CAMETOP432_BITLEN) {
```

Comment:
Maybe you could use https://github.com/merbanan/rtl_433/blob/9c117521f0570c86eda651c1237c3d796bfbb62d/include/bitbuffer.h#L102-L104

with for example a minimum of 2 rows (or more, but keep in mind that signal quality may vary). This would reduce false positives a lot, I think. At least, the comment above mentions that the packet is sent 4 times.

Reply:
I tried this method at first, but it did not work. Will do an other try.

Reply:
Yeah, since I tried to do the flex decoder and that failed, I don't think it's possible to get multiple rows.

Reply:
The packet gap is limited by "PD_MAX_GAP_RATIO times the longest pulse", i.e. gap_limit bigger than long_limit*10 won't work.